src/main/java/com/intel/gkl/compression/IntelDeflater.java

```diff
@@ -169,6 +169,12 @@ public void finish() {
      */
     @Override
     public int deflate(byte[] b, int off, int len ) {
+        if(b == null) {
+            throw new NullPointerException("Input is null");
+        }
+        if(len <= 0) {
+            throw new NullPointerException("Input buffer length is less or equal zero.");
+        }
 
         return deflateNative(b, len);
```

Comment:
`off` not passed to native function

Reply:
Should be?

Reply:
Just calling out for disscusion on L171.

Reply:
As discussed to day need to check if " off != 0 "  and I would vote to print an error message and return IllegalArgumentException.   This code only works if off==0 and that is not expected so lets print a message and return without doing any work.